corehq/apps/cleanup/management/commands/couch_migrate.py

```diff
@@ -32,10 +40,42 @@ def handle(self, *args, **options):
             migration_config = MigrationConfig.wrap(json.loads(f.read()))
             if options['replicate']:
                 _replicate(migration_config)
+            if options['copy']:
+                _copy(migration_config)
             if options['check']:
                 _check(migration_config)
 
 
+def _copy(config):
+    # unfortunately the only couch view we have for this needs to go by domain
+    # will be a bit slow
+    database = Domain.get_db()
+    assert database.uri == config.source_db.uri, 'can only use "copy" with the main HQ DB as the source'
+    domain_names = Domain.get_all_names()
+    for domain in domain_names:
+        for doc_type in config.doc_types:
+            ids_of_this_type = [row['id'] for row in database.view(
+                'domain/docs',
+                startkey=[domain, doc_type],
+                endkey=[domain, doc_type, {}],
+                reduce=False,
+                include_docs=False,
+            )]
+            for id_group in chunked(ids_of_this_type, 500):
+                docs = get_docs(database, id_group)
+                new_revs = dict([
+                    (row['id'], row['value']['rev'])
+                    for row in config.dest_db.view('_all_docs', include_docs=False)
```

Comment:
Would it not be better to do this once outside the loops rather than for every group?